### PR TITLE
Enable Method Configs with same name and different namespace

### PR DESCRIPTION
**Pull request by jmthibault79**
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: **Assign PR to** 
- [x] : does a once-over of the PR
- [x] : Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **LR**: sign off, **assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by jmthibault79** [comment]
My preference is for `idField` to be an `Option` but it's included here as a separate commit in case of disagreement.

**Comment by helgridly** [comment]
I'm gonna anoint myself as LR so we can get moving on this.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 1]
```diff
@@ -16,7 +21,24 @@ trait Attributable {
...
 trait DomainObject {
   //the name of a field on this object that uniquely identifies it relative to any graph siblings
-  def idField: String
+  protected def idField: Option[String]
+
+  // a function to uniquely identify this object relative to any graph siblings
+  // by default, it checks the idField for equality
+  def idFilterFn(tpe: Type)(vertex: Vertex): Boolean = {
+    this.idField match {
+      case Some(id) => getDomainObjectIdValue(tpe, this) == vertex.getProperty(id)
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: I have serious reservations about this.
1. I don't like that our serialization code is leaking out of GraphDAO classes.
2. Not only is the serialization code leaking, but the specifics of _how_ fields are serialized are leaking. Even if the answer to #1 is "tough crap", I think `vertex.getProperty()` should be replaced by a call to `GraphDAO.loadProperty()` in order to keep the implementation details encapsulated.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 2]
```diff
@@ -109,7 +131,12 @@ case class MethodConfiguration(
                    ) extends DomainObject {
   def toShort : MethodConfigurationShort = MethodConfigurationShort(name, rootEntityType, methodRepoMethod, namespace)
   def path( workspaceName: WorkspaceName ) = workspaceName.path+s"/methodConfigs/${namespace}/${name}"
-  def idField = "name"
+
+  // a custom idFilterFn is required because two fields are necessary to identify
+  def idField = None
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: I'm not sure what `idField` being an option achieves here. It's checked in the default implementation of `idFilterFn`, but tripping that error would require that you set `idField = None` and fail to override `idFilterFn`. Surely you're more likely to do the opposite if you're in a rush - and that'd work fine?

Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: Makes sense.  I'll revert.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala (hunk 1 above)
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: Good points.

I'm throwing this out for a much cleaner and simpler solution.

**Comment by dvoet** [comment]
:+1:

**Comment by helgridly** [comment]
I can break this by setting `idFields = Seq("SomeFieldThatIsntAString").`

I still think the `domainObjectFilterFn` should use `loadProperty` not `vertex.getProperty`.

**Pull request merged**